Watchman.AwsResources/Services/AutoScaling/AutoScalingGroupAlarmDataProvider.cs

```diff
@@ -21,17 +33,62 @@ private static string GetAttribute(AutoScalingGroup resource, string property)
             }
         }
 
-        public decimal GetValue(AutoScalingGroup resource, string property)
+        public async Task<decimal> GetValue(AutoScalingGroup resource, AutoScalingResourceConfig config, string property)
         {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+            
             switch (property)
             {
                 case "GroupDesiredCapacity":
-                    return resource.DesiredCapacity;
+                    return await GetGroupDesiredValue(resource, config);
             }
 
             throw new Exception("Unsupported property name");
         }
 
+        private async Task<decimal> GetGroupDesiredValue(AutoScalingGroup resource, AutoScalingResourceConfig config)
+        {
+            if (config.InstanceCountIncreaseDelayMinutes == null)
+            {
+                return resource.DesiredCapacity;
+            }
+
+            var delaySeconds = config.InstanceCountIncreaseDelayMinutes.Value * 60;
```

Comment:
Do we prefer `InstanceCountIncreaseDelayMinutes` or `InstanceCountIncreaseDelaySeconds` ?
i.e. Is there ever a need to fine tune it to fractions of a minute?

Reply:
CloudWatch will only give you the metric at 1 minute intervals so this shouldn't be required.
I don't think the name is very good, if you can think of an improvement please say.

Reply:
I don't have a better name, and I don't mind the current one - at least it's descriptive.